OpenFoodFactsPower.user.js

```diff
@@ -838,9 +912,42 @@ content: " — ";
 
     }
 
+	//Copy data from the language specific ingredients_text to the ingredients_text in the hidden form so it can be poassed to the analyser
+	function Copydata(){
+		var lang = $('ul#tabs_ingredients_image > li.active').attr("data-language");
+        var pageLanguage = $("html").attr('lang');      // Get page language
 
+		//console.log("Lang:" + lang);
+		var cd = $("#ingredients_text_"+lang).val();
+		//console.log("Language Text:"+cd);
 
+        //Here we have to manipulate the language for regional languages
+        if(lang === 'ca'){lang = 'es-ca';} //Catalan
+        if(lang === 'en'){
+            if(pageLanguage === 'en'){
+                lang = 'uk';//English
+            }
+            else
+            {
+                lang = pageLanguage + '-en'; //English from source language page
+            }
+        }
 
+		//As target language can be different from the page language we have to create the full URL
+		var URL = "http:/" + lang + ".openfoodfacts.org/cgi/test_ingredients_analysis.pl";
```

Comment:
```suggestion
		var URL = "https://" + lang + ".openfoodfacts.org/cgi/test_ingredients_analysis.pl";
```

Reply:
Unfortunately neither the form.setAttribute or form.action work with https: - the call does however rediect to the off https page.